src/JoypadModule/src/Module.cpp

```diff
@@ -27,51 +27,58 @@ bool JoypadModule::configure(yarp::os::ResourceFinder &rf)
 
     // set the module name
     std::string name;
-    if(!YarpUtilities::getStringFromSearchable(rf, "name", name))
+    if (!YarpUtilities::getStringFromSearchable(rf, "name", name))
     {
         yError() << "[configure] Unable to get a string from searchable";
         return false;
     }
     setName(name.c_str());
 
+    // get the joypad type
+    if (!YarpUtilities::getStringFromSearchable(rf, "type", m_joypadType))
+    {
+        yError() << "[configure] Unable to get a string from searchable";
+        return false;
+    }
+
     // set the deadzone interval
-    if(!YarpUtilities::getNumberFromSearchable(rf, "deadzone", m_deadzone))
+    if (!YarpUtilities::getNumberFromSearchable(rf, "deadzone", m_deadzone))
     {
         yError() << "[configure] Unable to get a double from a searchable";
         return false;
     }
 
     // set the maximum value measured by the joypad
-    if(!YarpUtilities::getNumberFromSearchable(rf, "fullscale", m_fullscale))
+    if (!YarpUtilities::getNumberFromSearchable(rf, "fullscale", m_fullscale))
     {
         yError() << "[configure] Unable to get a double from a searchable";
         return false;
     }
 
     const std::string connectPortsTag = "connectPortsSeparately";
-    //connectPortsSeparately is present or set to true
+    // connectPortsSeparately is present or set to true
     m_connectPortsSeparately = rf.check(connectPortsTag) && (rf.find(connectPortsTag).isNull() || rf.find(connectPortsTag).asBool());
 
-    std::vector<std::pair<std::string, int&>> joypadInputs = {{"forwardAxis", m_forwardAxis},
-                                                              {"rotationAxis", m_rotationAxis},
-                                                              {"sideAxis", m_sideAxis},
-                                                              {"prepareButton", m_prepareButton},
-                                                              {"startButton", m_startButton},
-                                                              {"pauseButton", m_pauseButton},
-                                                              {"stopButton", m_stopButton},
-                                                              {"connectGoalButton", m_connectGoalButton},
-                                                              {"connectRpcButton", m_connectRpcButton},
-                                                              {"disconnectButton", m_disconnectButton}};
-    for (auto& input : joypadInputs)
+    std::vector<std::pair<std::string, int &>> joypadInputs = {{"forwardAxis", m_forwardAxis},
+                                                               {"rotationAxis", m_rotationAxis},
+                                                               {"sideAxis", m_sideAxis},
```

Comment:
Probably you can extract the first three and parse separately axis and buttons, so that you avoid parsing the only the buttons in the pedals

Reply:
This should be fixed after [651d8a7](https://github.com/robotology/walking-controllers/pull/188/commits/651d8a7ae03e901f897c6e31301aca8e20200ed9)